core/settings/qPref.cpp

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "qPref_private.h"
+#include "qPref.h"
+
+qPref::qPref(QObject *parent) : 
+	QObject(parent)
+{
+}
+qPref *qPref::instance()
+{
+static qPref *self;
```

Comment:
Sorry for being so annoying, but...
1) indentation
2) my point was to make it
`static qPref *self = new qPref;`

The reason: This is guaranteed by the compiler to be thread safe (which admittedly is certainly not an issue here). The compiler does all the locking magic, which the version with `if` does not.

@dirkhh: I don't see how this could ever produce two different `qPref` objects.